libgadget/domain.c

```diff
@@ -129,12 +164,16 @@ void domain_decompose_full(DomainDecomp * ddecomp)
     static DomainDecompositionPolicy policies[16];
     static int Npolicies = 0;
 
+    /*Number of tasks to decompose to*/
+    int NTask;
+    MPI_Comm_size(MPI_COMM_WORLD, &NTask);
```

Comment:
Can we save NTask, ThisTask and MPI_Comm as mebers of ddecomp?

Most of the functions require consistent values of these variables.

Reply:
Another possible convention is that any function that needs MPI, pass a MPI_Comm. Getting ThisTask and NTask from MPI_Comm is fast, though we may want a helper macro to reduce the code clutter.

Reply:
This is a good idea. It would also let us hypothetically use MPI communicators for some future domain feature.